make/test.mk

```diff
@@ -94,7 +94,7 @@ ifeq ($(E2E_REPO_PATH),"")
 	# delete to have clear environment
 	rm -rf ${E2E_REPO_PATH}
 	# clone
-	git clone https://github.com/codeready-toolchain/toolchain-e2e.git --depth 1 ${E2E_REPO_PATH}
+	git clone https://github.com/codeready-toolchain/toolchain-e2e.git ${E2E_REPO_PATH}
```

Comment:
Just wondering, instead of taking a whole clone, can we fetch required commits of master till the point where we can merge and There might not be a ref that we're looking for.
```
while [ -z $( git merge-base master feature ) ]; do     
    git fetch -q --deepen=1 origin master feature;
done
```

we can start from `--deepgen=10`, some big number. if it's saving some time for us, then maybe worth trying this wdyt @MatousJobanek

Reply:
I was thinking about it too, but for the sake of keeping it simple (the logic is complex enough), I decided to clone the whole master... The reason was that I thought that the clone part is just a minor part of the whole e2e execution and doesn't take to much time...
anyway, as you are already asking to do that, let me try it ;-)

Reply:
@dipak-pawar check this 3cba5cc

Reply:
It seems to get too complicated :) I would rather stick with the simplest clone/fetch the whole history for the sake of simplicity.

Reply:
Yeah, I figured out that maintaining the lazy fetch and the loop is really painful and very complicates the code. I'm gonna stick with the simple version.